Web/tues_pairs/src/components/Pages/ChangeForms/emailChange.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { Component } from 'react';
+import { compose } from 'recompose';
+import { withRouter, Link } from 'react-router-dom';
+import { withAuthorization } from '../../Authentication';
+import { withFirebase } from '../../Firebase';
+import { withCurrentUser } from '../../Authentication/context';
+import * as ROUTES from '../../../constants/routes';
+import { Card, FormControl, Button, Form, Modal } from 'react-bootstrap';
+import log from '../../../constants/logger';
+
+class EmailChangeFormBase extends Component {
+    constructor(props) {
+        super(props);
+        
+        this.state = {
+            email: '',
+            error: '',
+            password: '',
+            modalShow: false,
+            setModalShow: false,
+            currentUser: Object
+        };
+    }
+
+    onChange = event => {
+        this.setState({ [event.target.name]: event.target.value });
+    }
+
+    onSubmit = () => {
+        const { email, password } = this.state;
+        let credential;
+        this.props.firebase.getCurrentUser()
+        .then(currentUser => {
+            this.setState({ currentUser });
+        }).then(() => {
+            credential = this.props.firebase.getCredentials(this.props.authUser.email, password);
+        }).then(() => {
+            this.state.currentUser.reauthenticateWithCredential(credential).then(() => {
+                this.props.firebase.db.collection("users").doc(this.props.authUser.uid).set({
+                    email: email,
+                }, {merge: true}).then(() => {
+                    this.props.firebase.doEmailUpdate(email).then(() => {
+                        log.info("Updated current user changed his email inside Email Change page!");
+                        this.props.history.push(ROUTES.EDIT_PERSONAL_INFO);
+                    }).catch(error => {
+                        log.error(error);
+                        this.setState({ error, modalShow: false });
+                    })
+                });
+            }).catch(error => {
+                log.error(error);
+                this.setState({ error, modalShow: false });
+            });
+        });
+    }
+
+    render() {
+        const{ error, email, modalShow } = this.state;
+
+        const isInvalid = email === '' ||
+        !email.includes('@');
```

Comment:
Nice validation

Reply:
This will be changed.

Reply:
I will add a custom validator for the email.